src/config/config.ts

```diff
@@ -0,0 +1,360 @@
+    },
+    redirectUri: {
+      doc: "URL to redirect to after authentication with GitHub",
+      env: "REDIRECT_URI",
+      format: "required-string",
+      default: null,
+    },
+    branchRef: {
+      doc: "Git branch to use for saving modifications to site",
+      env: "BRANCH_REF",
+      format: "required-string",
+      default: "staging",
+    },
+    systemToken: {
+      doc: "GitHub access token to create repo",
+      env: "SYSTEM_GITHUB_TOKEN",
+      sensitive: true,
+      format: "required-string",
+      default: null,
+    },
+  },
+  dataDog: {
+    env: {
+      doc: "The DataDog environment",
+      format: ["production", "development", "local"],
+      env: "DD_ENV",
+      default: "local",
+    },
+    service: {
+      doc: "The DataDog service",
+      env: "DD_SERVICE",
+      format: "required-string",
+      default: null,
+    },
+    tags: {
+      doc: "The DataDog tags",
+      env: "DD_TAGS",
+      format: "required-string",
+      default: null,
+    },
+  },
+  formSg: {
+    siteCreateFormKey: {
+      doc: "FormSG API key for site creation form",
+      env: "SITE_CREATE_FORM_KEY",
+      sensitive: true,
+      format: "required-string",
+      default: null,
+    },
+  },
+  postman: {
+    apiKey: {
+      doc: "Postman API key",
+      env: "POSTMAN_API_KEY",
+      sensitive: true,
+      format: "required-string",
+      default: "",
+    },
+    smsCredName: {
+      doc: "Postman SMS credential name",
+      env: "POSTMAN_SMS_CRED_NAME",
+      format: "required-string",
+      default: "",
+    },
+  },
+  cypress: {
+    e2eTestRepo: {
+      doc: "Name of the e2e test GitHub repository",
+      env: "E2E_TEST_REPO",
+      format: "required-string",
+      default: null,
+    },
+    e2eTestSecret: {
+      doc: "Secret for e2e tests",
+      env: "E2E_TEST_SECRET",
+      sensitive: true,
+      format: "required-string",
+      default: null,
+    },
+    e2eTestGithubToken: {
+      doc:
+        "GitHub access token for e2e tests. Replace with your own token and make sure the github user is in your local database",
+      env: "E2E_TEST_GH_TOKEN",
+      sensitive: true,
+      format: "required-string",
+      default: null,
+    },
+  },
+  database: {
+    dbUri: {
+      doc: "Database URI",
+      env: "DB_URI",
+      sensitive: true,
+      format: "required-string",
+      default: null,
+    },
+    dbMinPool: {
+      doc: "Minimum number of connections in the pool",
+      env: "DB_MIN_POOL",
+      format: "required-number",
+      default: 1,
+    },
+    dbMaxPool: {
+      doc: "Maximum number of connections in the pool",
+      env: "DB_MAX_POOL",
+      format: "required-number",
+      default: 10,
+    },
+    dbEnableLogging: {
+      doc: "Enable database logging",
+      env: "DB_ENABLE_LOGGING",
+      format: "required-boolean",
+      default: false,
+    },
+  },
+})
+
+// Perform validation
+config.validate({ allowed: "strict" })
+
+module.exports = config
```

Comment:
wa why mix commonJS exports and es2021 exports :(

Reply:
Hmm yea, I was trying use `import` syntax everywhere, but didnt work, had to use require in some places

Reply:
could i get more insight into this? it sounds p important

Reply:
@seaerchin oops replied in below thread
https://github.com/isomerpages/isomercms-backend/pull/646#discussion_r1127616701

Reply:
it works if we use named exports rather than default ._. i've a feeling this is because the transpilation doesn't map [cleanly](https://stackoverflow.com/questions/40294870/module-exports-vs-export-default-in-node-js-and-es6)...

if you don't mind, i can commit directly here

Reply:
@seaerchin sure go ahead

Reply:
done